tests/test_plugin.py

```diff
@@ -261,3 +263,186 @@ def test_crud_for_navigation_plugin_when_versioning_disabled(self):
 
         # Now delete the plugin from the page and assert
         self._delete_nav_plugin_and_assert(new_placeholder, new_plugin)
+
+    def test_menu_cache_invalidate_after_menuitem_creation(self):
+        # NOTE: This test is based on a similar one from django-cms:
+        # https://github.com/divio/django-cms/blob/2daeb7d63cb5fee49575a834d0f23669ce46144e/cms/tests/test_plugins.py#L160
+
+        # Set up a versioned page with one placeholder
+        page_content = factories.PageContentWithVersionFactory(
+            language=self.language, version__created_by=self.get_superuser()
+        )
+        placeholder = factories.PlaceholderFactory(source=page_content)
+        menu_content = factories.MenuContentWithVersionFactory()
+        child = factories.ChildMenuItemFactory(parent=menu_content.root)
+        grandchild = factories.ChildMenuItemFactory(parent=child)
+
+        # Add nav plugin to placeholder
+        self._add_nav_plugin_and_assert(
+            placeholder, menu_content.menu, "menu/menu.html"
+        )
+
+        # Now publish the page content containing the plugin,
+        # so the page can be viewed
+        version = page_content.versions.get()
+        version.publish(self.get_superuser())
+
+        # Making sure there is no cachekey existed before rendering page
+        cache_key = CacheKey.objects.all().count()
+        self.assertEqual(cache_key, 0)
+
+        # And view the page
+        page_url = page_content.page.get_absolute_url()
+        response = self.client.get(page_url)
+
+        # Rendering should generate cachekey object
+        cache_key = CacheKey.objects.all().count()
+        self.assertEqual(cache_key, 1)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(child.title, str(response.content))
+        self.assertIn(grandchild.title, str(response.content))
+
+        # Create further menu items which covers draft state
```

Comment:
What does "covers draft state" mean? And why does the draft state matter in this test? I thought this line was here simply because you want to check that when the menu version is modified, the cache clearing fires up?